### PR TITLE
Quirk: Light Sleeper

### PR DESCRIPTION
**Pull request by V2LenKagamine**
## About The Pull Request
_**HEY! GIVE ME FEEDBACK! I'm here to put out an idea, you're here to help me refine it into something worthy of being in the game!**_

This simple pull request does 2 things:
Adds the quirk ~~"Insomniac"~~ "Light Sleeper" reducing sleep and unconscious times by 25%
Changes "Heavy Sleeper" to no longer be random, and always increase sleep and unconscious times by 25%

## Why It's Good For The Game
Ever wonder why Heavy Sleeper didn't have a POSITIVE variant? I did.

I think we could use more quirks, while I would prefer them to be more game changing, the community appears to only want RP related quirks. So I made a less impactful quirk. Now you can be even more of a snowflake!

## Changelog
:cl:
add: Added Quirk: Light Sleeper
tweak: "Heavy Sleeper" now always increases Sleep and Unconscious times by 25% instead of being between 25% and 30%.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Make them mutually exclusive; taking both heavy sleeper AND insomniac gives you a 6.25% reduction in knockout times for a net quirk cost of 0 points.

**Comment by RaveRadbury** [comment]
> Make them mutually exclusive; taking both heavy sleeper AND insomniac gives you a 6.25% reduction in knockout times for a net quirk cost of 0 points.

Alternatively rebalance them so you get a 6.25% _increase_ and greedy fools are punished.

**Comment by anconfuzedrock** [comment]
alternatively don't add this, this is too close to the "no combat quirks" thing which I May not agree with but every single fucking powergamer will get reduced sleep times and, hell, everyone, since being able to wake up faster is so convenient.

**Comment by tralezab** [comment]
So weird seeing "Insomniac" as a good perk and "Heavy Sleeper" as a bad one

**Comment by ATH1909** [comment]
> alternatively don't add this, this is too close to the "no combat quirks" thing which I May not agree with but every single fucking powergamer will get reduced sleep times and, hell, everyone, since being able to wake up faster is so convenient.

Then what the hell _are_ we supposed to spend our quirk points on? Everyone seems to protest the addition of anything that might be worth taking a negative quirk (other than light drinker; that's practically a freebie) for.

**Comment by Ratvar** [comment]
As Oranges said, "Quirks are meant to be RP flavouring with minor in game impacts", not for buffing your character up. Yes, not for being *worth* to take, but for being snowflake. Like, can you say why you wouldn't take this trait?

File: code/modules/mob/living/carbon/human/status_procs.dm [hunk 1]
```diff
@@ -18,12 +18,16 @@
 /mob/living/carbon/human/Unconscious(amount, updating = 1, ignore_canunconscious = 0)
 	amount = dna.species.spec_stun(src,amount)
 	if(has_trait(TRAIT_HEAVY_SLEEPER))
-		amount *= rand(1.25, 1.3)
+		amount *= 1.25
+	if(has_trait(TRAIT_LIGHT_SLEEPER && !TRAIT_HEAVY_SLEEPER))
```
Comment (Mickyan) [inline code/modules/mob/living/carbon/human/status_procs.dm]: we have quirk_blacklist for this

**Comment by anconfuzedrock** [comment]
just because you don't like the current quirks doesnt mean they're useless. You're probably a validhunter who plays a byond game for the combat, so I recommend empath: that way you can perform disgusting antag checks by forcefeeding people urinal cakes and seeing if that makes their sanity go down too much. Otherwise, climbing tables faster or being able to hide in a locker while handcuffed when you're being dragged off to brig for smashing random windows with a fire axe and the officer slips and you run off faster than your mother to an abortion clinic in an alternate reality where she can see what a disappointment you're going to be, could have use. Or- here's an idea- don't take quirks at all if you don't like the upsides, nobody is forcing you to take the downsides.

**Comment by RaveRadbury** [comment]
The funny thing is that Heavy Sleeper is already a power-game quirk because a lot of people believe that once you've gone horizontal the fight is over anyway. So in their minds it's already free points.

File: code/modules/mob/living/carbon/human/status_procs.dm (hunk 1 above)
Comment (dcaine00) [reply code/modules/mob/living/carbon/human/status_procs.dm]: Sorry, Im a bit new to coding and the codebase, and this was the easiest and quickest solution I could think of.

**Comment by OverlyGeneric** [comment]
> The funny thing is that Heavy Sleeper is already a power-game quirk because a lot of people believe that once you've gone horizontal the fight is over anyway. So in their minds it's already free points.

It also increases the time for regenerative coma to heal you. I think.

**Comment by Militaires** [comment]
look in quirks.dm subsystem folder, you'll find the blacklist there

File: code/modules/mob/living/carbon/human/status_procs.dm (hunk 1 above)
Comment (V2LenKagamine) [reply code/modules/mob/living/carbon/human/status_procs.dm]: Fixed!

**Comment by PKPenguin321** [comment]
@optimumtact combat oriented quirk detected

**Comment by optimumtact** [comment]
what things in combat make you go to sleep?

**Comment by PKPenguin321** [comment]
off the top of my head:
syringe guns with sleep chems (very common)
sleep chems in general
abductor sleep prod
boxing gloves KO punch

**Comment by Militaires** [comment]
honestly if someone is gonna kill you when you're asleep a 25% shorter time wont make a difference, so you can actually just not consider it as a combat quirk

**Comment by Shadowflame909** [comment]
How do you get up 25% faster anyways when Morphine is inside of you. Wouldn't the chemical be forcing you asleep anyways?

**Comment by anconfuzedrock** [comment]
@optimumtact most importantly explosions though, especially fireballs. this would make it more feasible to get up from a fireball or welderbomb and put yourself out.

Either way, this quirk is absurdly good and it will be unless it's like 3 points at minimum, I'm shocked and apalled this guy looked at the feedback and thought "yep what this needs is a name change." Like I couldn't imagine shitting out this many prs with so much effort put into them yet so little thought.

**Comment by Tlaltecuhtli** [comment]
sleeping from chems isnt actually important as you ll be asleep as long as you have the chem in body (it only affects the last few seconds after it depletes )

real things affected by heavy sleeper/insomniac

cryo torture pod

rng explosion knockdown which i think should be removed

**Comment by anconfuzedrock** [comment]
I swear to god if I'm a cook and this fucking assistant with fast tableclimbing wakes up even faster after they get knocked out with CC and pours acid on my spaghetti I am going to lose it

**Comment by ExcessiveUseOfCobblestone** [comment]
The fact unconscious and sleep are 2 different procs leads me to believe at least 1 has combat usage if not both.

**Comment by ExcessiveUseOfCobblestone** [comment]
https://github.com/tgstation/tgstation/blob/c73cf825be70acde4580410cd4f11cd19fe72248/code/game/gamemodes/cult/cult.dm#L110-L111

https://github.com/tgstation/tgstation/blob/c73cf825be70acde4580410cd4f11cd19fe72248/code/game/gamemodes/cult/cult.dm#L121-L122

https://github.com/tgstation/tgstation/blob/f642c66986fee91548f2110185bc2057ee0e2b9d/code/modules/spells/spell_types/hivemind.dm#L201

https://github.com/tgstation/tgstation/blob/55a768fef109af873291a4383e1b735a5e5969c5/code/modules/hydroponics/grown/nettle.dm#L113

just a few

**Comment by anconfuzedrock** [comment]
!!Mindswap!!

**Comment by necromanceranne** [comment]
CQC knockout. But frankly if someone has you in CQC stunlock hell then likely you're already dead.

**Comment by ExcessiveUseOfCobblestone** [comment]
The implication is that if they can knock you down they can follow up, which isn't always true (at least to the point where this is negligible for considering this combat-oriented).

**Comment by anconfuzedrock** [comment]
I greatly struggle to think of a situation where you're knocked out that isn't combat oriented besides *sleep which is lol you're trying to sleep

**Comment by CRITAWAKETS** [comment]
Honestly, i would not have this quirk because its useless. Most situations where you get knocked out are you being unconcious, not asleep. This doesnt help agaisnt sleep chems, infact it can make it worse with some doses because you can go back to sleep a second time sooner.

**Comment by ExcessiveUseOfCobblestone** [comment]
Basically a stun-modifying quirk which is not allowed.

**Pull request closed without merging**